### PR TITLE
fix: make page[] param wrap consistent

### PR DESCRIPTION
**Pull request by dugsmith**
This is an opinionated fix to issue #347. It makes json_api_client handle pagination query params consistently by wrapping `page[]` around them in all places they are used. Of course, this can be overridden in custom paginators.

### CODE REVIEW

**Comment by dugsmith** [comment]
@gaorlov: do you know when you might have a chance to consider this?

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1 @@
+ruby 2.6.3
```
Comment (gaorlov) [inline .tool-versions]: Do we need this here?

File: .gitignore [hunk 2]
```diff
@@ -24,3 +24,6 @@ Gemfile.lock
 *.gemfile.lock
 
 /coverage
+
+*.iml
+.idea/
```
Comment (gaorlov) [inline .gitignore]: What are these?

File: .tool-versions (hunk 1 above)
Comment (dugsmith) [reply .tool-versions]: No, I can take that out. I'll add it to .gitignore.

File: .gitignore (hunk 2 above)
Comment (dugsmith) [reply .gitignore]: These are IntelliJ/RubyMine config files/directories.

**Comment by dugsmith** [comment]
@gaorlov Thank you for your feedback! You're right, it is a breaking change. I will look into making this an alternate behavior while leaving the current behavior the default. That will make migration easier, for sure.

**Comment by dugsmith** [comment]
@gaorlov I've updated this to use a `NestedParamPaginator` as the one that has consistently nested behavior across all use-cases. The original behavior is the default. How does this look?

**Comment by dugsmith** [comment]
Thanks, @gaorlov! I've updated README and CHANGELOG. I assumed you'd want to increment the version when you merge, but let me know if you need me to add that commit too.

**Comment by dugsmith** [comment]
@gaorlov Did you want me to bump the gem version before you merge this to master?

**Comment by gaorlov** [comment]
@dugsmith I'll bump the version in master. I'll merge it on and ship the green in the next hour or so. Thank you for your work!

**Comment by dugsmith** [comment]
@gaorlov: Excellent! Thank you for shepherding this great project!

**Comment by gaorlov** [comment]
@dugsmith `1.13.0` is now live. Thank you for your work and for accommodating my requests. I really appreciate it.

**Comment by dugsmith** [comment]
> @dugsmith `1.13.0` is now live. Thank you for your work and for accommodating my requests. I really appreciate it.

@gaorlov Thank you for turning this around so quickly! It helps us a lot! :-)

**Pull request merged**